questionnaire/questionnaire-dal.js

```diff
@@ -177,6 +177,34 @@ function questionnaireDAL(spec) {
         return result;
     }
 
+    async function getQuestionnaireMetadata(query) {
+        let result;
+        try {
+            if (query?.filter?.userId) {
+                result = await db.query(
+                    `SELECT id, questionnaire -> 'meta' -> 'questionnaireDocumentVersion' AS "questionnaire-version", created, modified, submission_status, questionnaire -> 'answers' -> 'user' -> 'user-id' AS "user-id" FROM questionnaire WHERE questionnaire -> 'answers' -> 'user' ->> 'user-id' = $1`,
+                    [query.filter.userId]
+                );
+                if (result.rowCount === 0) {
+                    throw new VError(
+                        {
+                            name: 'ResourceNotFound'
+                        },
+                        `User "${query.filter.userId}" not found`
+                    );
+                }
+            } else {
+                result = await db.query(
+                    "SELECT id, questionnaire -> 'meta' -> 'questionnaireDocumentVersion' AS \"questionnaire-version\", created, modified, submission_status, questionnaire -> 'answers' -> 'user' -> 'user-id' AS \"user-id\" FROM questionnaire"
+                );
+            }
+        } catch (err) {
+            throw err;
+        }
+
+        return result;
```

Comment:
.map() this to remove all other redundant data in the response. Just return an array of objects